ansible/roles/google_cloud/files/function/convert_glossis_netcdf_to_geotiff.py

```diff
@@ -0,0 +1,256 @@
+        for i, (face, count) in tqdm.tqdm(enumerate(zip(face_coordinates, counts)), desc='grid->features'):
+            poly = shapely.geometry.Polygon(face[:count].tolist())
+            # polys.append(poly)
+            if meridian.intersects(poly):
+                # split_poly = shapely.ops.split(poly, meridian)
+                # for poly in split_poly.geoms:
+                #     ipdb.set_trace()
+                #     polys.append(poly)
+                old_coords = face[:count].tolist()
+                shift_coords = []
+                for x in old_coords:
+                    shift_coords = shift_coords + [[x[0] + 360, x[1]]]
+                poly = shapely.geometry.Polygon(shift_coords)
+            polys.append(poly)
+
+        remove(domain_file)
+        break
+
+    collection = geojson.FeatureCollection(features=features,
+                                           properties=metadata
+                                           )
+    save_path = path + value_string + '_' + time.strftime("%Y%m%d_%H%M%S")
+
+    # with open(save_path +'.geojson', 'w') as f:
+    #     geojson.dump(collection, f)
+
+    # rasterize the feature collection to WGS84 raster with nx x ny pixels
+    rasters = []
+    for i, layer in enumerate(layer_types[value_string]):
+        shapes = [
+            (feature.geometry, feature.properties[layer])
+            for feature
+            in collection.features
+        ]
+
+        raster = rasterio.features.rasterize(
+            shapes,
+            out_shape=(nx, ny),
+            transform=transform,
+            fill=NODATA
+        )
+        rasters.append(raster)
+
+    # make model boundary mask as raster
+    raster = rasterio.features.rasterize(
+        ((poly, i) for (i, poly) in enumerate(polys)),
+        out_shape=(nx, ny),
+        transform=transform,
+        fill=NODATA
+    )
+
+    rasters.append(raster)
+
+    dst = rasterio.open(
+        save_path + '_interpolated.tif',
+        'w',
+        driver='GTiff',
+        height=rasters[0].shape[0],
+        width=rasters[0].shape[1],
+        count=len(rasters),
+        dtype='float64',
+        crs='epsg:' + str(dst_epsg),
+        transform=transform
+    )
+    for i, layer in enumerate(rasters):
+        grid_x, grid_y = np.mgrid[0:nx, 0:ny]
+        values = layer[layer != NODATA]
+        grid_x_val = grid_x[layer != NODATA]
+        grid_y_val = grid_y[layer != NODATA]
+        if i <= 1:
+            points = np.column_stack([grid_x_val, grid_y_val])
+            interpolated = griddata(points, values, (grid_x, grid_y), method='linear')
+            dst.write_band(i + 1, interpolated)
+        else:
+            is_grid = np.zeros((nx, ny), dtype='float64')
+            for x_, y_ in zip(grid_x_val, grid_y_val):
+                is_grid[x_, y_] = 1.0
+
+            dst.write_band(i + 1, is_grid)
+    time_meta = {
+        "system:time_start": time.strftime("%Y%m%d %H%M%S"),
```

Comment:
The metadata with ":" character is not allowed by general metadata. This needs to be in Unix epoch (YYYY-MM-DDThh:mm:ss) for GEE upload. "system:time_start" can be specified in that upload rather than here.